opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -275,6 +289,30 @@ def set_status(self, status: trace_api.Status) -> None:
             return
         self.status = status
 
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+    ) -> None:
+        """Ends context manager and calls `end` on the `Span`."""
+
+        if self.status is None and self._auto_update_status:
+            if exc_val is not None:
+
+                if isinstance(exc_val, StatusError):
+                    self.set_status(
+                        Status(exc_val.canonical_code, exc_val.description)
+                    )
+
+                else:
+                    self.set_status(Status(StatusCanonicalCode.UNKNOWN))
+
+            else:
+                self.set_status(Status(StatusCanonicalCode.OK))
+
+        super(Span, self).__exit__(exc_type, exc_val, exc_tb)
```

Comment:
I think the result should be the same either way. Just because of personal preference I have used `super` still, but removed the arguments as you indicated.